### PR TITLE
Move `overlay` planning to`ExprPlanner`

### LINKED ISSUE #11223: Implement `ExprPlanner` for `sql_overlay_to_expr`
### Is your feature request related to a problem or challenge?

As discussed in https://github.com/apache/datafusion/issues/11207 The idea is to move hardcoded the `sql_overlay_to_expr`  operator to function in the parser step, which is currently handled in the optimizer step. We need to implement user-defined parser mechanisms to enable more flexible SQL planning.

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dharanad**
## Which issue does this PR close?

Closes #11223

## Rationale for this change

Part of #11207

## Are these changes tested?

Existing test cases
```
cargo test
```

## Are there any user-facing changes?

No

### CODE REVIEW

File: datafusion/sql/src/expr/mod.rs [hunk 1]
```diff
@@ -193,7 +193,7 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                     }
                 }
 
-                not_impl_err!("Extract not supported by UserDefinedExtensionPlanners: {extract_args:?}")
+                not_impl_err!("Extract not supported by ExprPlanner: {extract_args:?}")
```
Comment (alamb) [inline datafusion/sql/src/expr/mod.rs:196]: 👍 

Thank you -- this is a nice drive by cleanup

**Review by alamb — APPROVED** [review]
Thank you @dharanad 🙏

**Comment by dharanad** [comment]
> Thank you @dharanad 🙏 

Thank for your time and review @alamb . Apologies for the delay, busy with personal last couple of days.

**Comment by alamb** [comment]
> > Thank you @dharanad 🙏
> 
> Thank for your time and review @alamb . Apologies for the delay, busy with personal last couple of days.

I *totally* understand. Everything you can do is appreciated. Thanks again

**Pull request merged**